### ISSUE TITLE
forward alarms logs to syslog

### ISSUE THREAD

**Issue by l2isbad** [firehol/netdata#1194]
The only alarm logs  we have are alarm-notify.sh logs  in error.log (besides logs at web ui)
I think we need one more recipient in alarm-notify.sh - syslog.
It would be very useful not only for active but for proactive monitoring (Im using kibana alot for proactive monitoring)

**Comment by ktsaou** [comment]
hm... should this include also silent alarms? I mean, the dashboard shows notifications as they happen and includes all alarms, while `alarm-notify.sh` is sending a subset of the alarms and it sends them with a delay.

I think sending all the alarms to syslog is better.
Right?

**Comment by shadycuz** [comment]
Alarms and logging of the netdata-update.sh by default. Recently had updates fail and didn't know why, I ended up just changing how I called the script from cron to >> log

**Comment by l2isbad** [comment]
I was thinking about remote syslog so only alarms for which it is specified in the configuration (kinda to: syslog1; role_recipients_syslog[syslog1]="1.1.1.1:514"

**Comment by ktsaou** [comment]
well, if you do it in `alarm-notify.sh` the syslog logs will have hysteresis applied, and all the `silent` ones will not be there at all.

It is relatively easy to do it at the netdata daemon, and also relatively easy to do it at `alarm-notify.sh`.
So, which one?

Vote:

- 👍  for embedding this in netdata (all alarms will be sent to syslog, as they happen, including the silent ones, much like the dashboard HTML5 notifications - limit: these will be sent to the system syslog)

-  👎  for `alarm-notify.sh` (only alarms that sent notifications will be sent to syslog, much like the emails - these could be sent to any syslog server, privided that the command `logger` support the option `--server`)

**Comment by l2isbad** [comment]
why there is no "both" option?)
local and remote logging are both good to have

**Comment by gamefundas** [comment]
Thanks @ktsaou for this wonderful tool. We just started using Netdata for monitoring our Ec2 instances and this tool has been very insightful and valuable. One of the things we do in our cluster as a notification pipeline is to send all alerts to syslog for processing to emails and incident management tools. If there is just a way to add syslog as an option to alarm-notify that works too as we don't want everything to end up as incidents, only the meaningful ones so post hysteresis works well in our case.

Is there an easy way for me to build this option in the current alarm-notify.sh itself?

**Comment by ktsaou** [comment]
Yes, you can. Just use the `logger` command. You can modify `alarms-notify.sh` to do this. Just follow, for example `pushover` or `pushbullet` to provide `syslog` (copy & paste mainlym with some edits).

@l2isbad suggested to use the destination address as the syslog server to send send the messages to, which would be nice if you could do it.

Once you are done, make a PR and I'll join to check it.

**Comment by gamefundas** [comment]
Sure, let me give this a shot and see if I can get it to work reasonably well.

**Comment by ktsaou** [comment]
nice!

**Comment by sLuvpreet33** [comment]
Has this been achieved ? Can I send alarm logs, and also other logs to rsyslog ? If yes, then please give a little hint.

In my /etc/netdata/health_alarm_notify.conf , I don't see the option to send logs to syslog.

**Comment by ktsaou** [comment]
In `netdata.conf` set `error log = syslog`. The same can be done for `access log` and `debug log`.

**Comment by sLuvpreet33** [comment]
Thanks for this,

I am planning to send logs to elasticsearch via rsyslog, so I will need to view stats about rsyslog.
Can netdata monitor rsyslog ? the number of packets sent and received via respective protocols  ?

**Comment by ktsaou** [comment]
> Can netdata monitor rsyslog ? the number of packets sent and received via respective protocols ?

not currently, but this is a nice idea to be added to #2174

**Comment by sLuvpreet33** [comment]
Hey @ktsaou actually rsyslog is monitored in systemd services except for the network info.

How can I change the order of applications appearing in the chart legend ? I want rsyslog, grafana-server and netdata at the top, but they are down in the order, therefore, everytime I have to scroll the legend much before seeing them.

**Comment by ktsaou** [comment]
> How can I change the order of applications appearing in the chart legend ? I want rsyslog, grafana-server and netdata at the top, but they are down in the order, therefore, everytime I have to scroll the legend much before seeing them.

You can't currently, but you can resize the chart verically, and netdata will remember its size (it will be saved at your browser, so every time to look at it with the same browser, the size will be the one you last used).

There is also a request for customizing the order of dimensions in charts #1566 - of course every need is different (for example, others need to have first the dimensions with the most visible volume).

**Comment by gamefundas** [comment]
Most central logging solutions have the ability to source system events/logs directly from syslog as the format is a standard one and integrations are usually in place via log agents. 

Now if Netdata alerts can be sent to syslogs the whole integration becomes lot easier. As existing alerting mechanisms running off the central logging will consume the Netdata event simply as any other events. The only other thing I have needed in the past is the ability to tag the alert message with some extra context so that it can be categorized and processed by alerting rules. 

You probably meant Netdata should not be in the business of managing logs which makes perfect sense as there are sophisticated solutions for those.

**Comment by Ferroin** [comment]
This might be closable.  Netdata's master branch currently supports sending alarm notifications to syslog using arbitrary user specified templates that can contain any of the variables accessible to alarm-notify.sh.

Relevant PR is #3599.
